CleverTap/Runtime/Native/UnityNativeWrapper/Models/UnityNativeAppInboxPersistence.cs

```diff
@@ -0,0 +1,323 @@
+
+            List<string> ids = GetAllMessageIds();
+
+            for (int i = 0, count = ids.Count; i < count; i++)
+            {
+                string id = ids[i];
+                string msg = GetMessage(id);
+
+                if (!string.IsNullOrEmpty(msg))
+                {
+                    dict[id] = msg;
+                }
+            }
+
+            return dict;
+        }
+
+        /// <summary>
+        /// Marks all stored messages as read.
+        /// </summary>
+        public static void MarkAllAsRead()
+        {
+            if (!IsInitialized())
+            {
+                return;
+            }
+
+            List<string> ids = GetAllMessageIds();
+
+            for (int i = 0, count = ids.Count; i < count; i++)
+            {
+                MarkAsRead(ids[i]);
+            }
+        }
+
+        /// <summary>
+        /// Marks only the specified message IDs as read.
+        /// </summary>
+        public static void MarkMessagesAsReadForIds(string[] ids)
+        {
+            if (!IsInitialized())
+            {
+                return;
+            }
+
+            for (int i = 0, count = ids.Length; i < count; i++)
+            {
+                MarkAsRead(ids[i]);
+            }
+        }
+
+        /// <summary>
+        /// Deletes all stored messages and their metadata (IDs and read flags).
+        /// </summary>
+        public static void DeleteAllMessages()
+        {
+            if (!IsInitialized())
+            {
+                return;
+            }
+
+            List<string> ids = GetAllMessageIds();
+
+            for (int i = 0, count = ids.Count; i < count; i++)
+            {
+                string id = ids[i];
+                _preferenceManager.DeleteKey($"{_msgPrefixKey}_{id}");
+                _preferenceManager.DeleteKey($"{_readPrefixKey}_{id}");
+            }
+
+            _preferenceManager.DeleteKey(_inboxIdsKey);
+        }
+
+        /// <summary>
+        /// Retrieves only the unread messages.
+        /// </summary>
+        /// <returns>A dictionary of unread messages keyed by their IDs.</returns>
+        public static Dictionary<string, string> GetUnreadMessages()
+        {
+            if (!IsInitialized())
+            {
+                return null;
+            }
+
+            var allMessages = GetAllMessages();
+            var unreadMessages = new Dictionary<string, string>();
+
+            foreach (var kvp in allMessages)
+            {
+                if (!IsRead(kvp.Key))
+                {
+                    unreadMessages[kvp.Key] = kvp.Value;
+                }
+            }
+
+            return unreadMessages;
+        }
+        #endregion
+
+        #region Private Static Methods
+
+        /// <summary>
+        /// Retrieves a list of all stored message IDs.
+        /// </summary>
+        /// <returns>A list of message IDs.</returns>
+        private static List<string> GetAllMessageIds()
```

Comment:
Consider returning a HashSet. Other method using the ids perform Contains and Remove which perform much better on a HashSet than a List.